src/main/java/datawave/microservice/authorization/postgres/PostgresDatawaveUserService.java

```diff
@@ -0,0 +1,73 @@
+package datawave.microservice.authorization.postgres;
+
+import static datawave.microservice.authorization.jsd.JsdDatawaveUserService.CACHE_NAME;
+
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+import org.springframework.cache.annotation.CacheConfig;
+import org.springframework.cache.annotation.CacheEvict;
+
+import datawave.microservice.cached.CacheInspector;
+import datawave.security.authorization.CachedDatawaveUserService;
+import datawave.security.authorization.DatawaveUser;
+import datawave.security.authorization.DatawaveUserInfo;
+import datawave.security.authorization.SubjectIssuerDNPair;
+
+/** A version of the {@link CachedDatawaveUserService} that returns results from the JSD Cache */
+@CacheConfig(cacheNames = CACHE_NAME)
+public class PostgresDatawaveUserService implements CachedDatawaveUserService {
+    public static final String CACHE_NAME = "datawaveUsers";
+    
+    private final PostgresDatawaveUserLookup jsdDatawaveUserLookup;
+    private final CacheInspector cacheInspector;
+    
+    public PostgresDatawaveUserService(PostgresDatawaveUserLookup jsdDatawaveUserLookup, CacheInspector cacheInspector) {
+        this.jsdDatawaveUserLookup = jsdDatawaveUserLookup;
+        this.cacheInspector = cacheInspector;
+    }
+    
+    @Override
+    public Collection<DatawaveUser> lookup(Collection<SubjectIssuerDNPair> dns) {
+        return dns.stream().map(jsdDatawaveUserLookup::lookupUser).collect(Collectors.toList());
+    }
+    
+    @Override
+    public Collection<DatawaveUser> reload(Collection<SubjectIssuerDNPair> dns) {
+        return dns.stream().map(jsdDatawaveUserLookup::reloadUser).collect(Collectors.toList());
+    }
+    
+    @Override
+    public DatawaveUser list(String name) {
+        return cacheInspector.list(CACHE_NAME, DatawaveUser.class, name.toLowerCase());
+    }
+    
+    @Override
+    public Collection<? extends DatawaveUserInfo> listAll() {
+        return cacheInspector.listAll(CACHE_NAME, DatawaveUser.class).stream().map(DatawaveUserInfo::new).collect(Collectors.toList());
```

Comment:
Are we concerned with the cacheInspector.listAll returning null? If it does it'd raise an uncaught exception when you try to call `.stream()`. Maybe call that, get the collection of listeners, null check, and then continue with either the stream or return an empty list.

Reply:
Splitting it out would also allow you to add logging!

Reply:
that is code provided to me by Adam, I honestly dont know.